UI/window-basic-main.cpp

```diff
@@ -3805,6 +3809,50 @@ QMenu *OBSBasic::AddScaleFilteringMenu(obs_sceneitem_t *item)
 	return menu;
 }
 
+QMenu *OBSBasic::AddBackgroundColorMenu(obs_sceneitem_t *item)
+{
+	QMenu *menu = new QMenu(QTStr("ChangeBG"));
+	QAction *action;
+	obs_data_t *privData = obs_sceneitem_get_private_settings(item);
```

Comment:
I think this is never released -- causing a memory leak.

One trick you can do is something like this:

```cpp
	QMenu *menu = new QMenu(QTStr("ChangeBG"));
	QAction *action;

	OBSData privData = obs_sceneitem_get_private_settings(item);
	obs_data_release(privData);

	obs_data_set_default_int(privData, "color-preset", 0);
	int preset = obs_data_get_int(privData, "color-preset");
```

I should probably add some constructor or sub-class of OBSData that allows it to pass the pointer without incrementing the reference.  Right now you have to release it right after because OBSData automatically increments the reference on construction.

Either that, or you can release it with `obs_data_release()` normally when you're finished with the data, C-style.